src/osd/osd_types.h

```diff
@@ -1380,7 +1380,16 @@ struct pg_pool_t {
   std::map<std::string, std::string> properties;  ///< OBSOLETE
   std::string erasure_code_profile; ///< name of the erasure code profile in OSDMap
   epoch_t last_change = 0;      ///< most recent epoch changed, exclusing snapshot changes
-
+  // If non-zero, require OSDs in at least this many different instances...
+  uint32_t peering_crush_bucket_count = 0;
+  // of this bucket type...
+  uint32_t peering_crush_bucket_barrier = 0;
+  // including this one
+  int32_t peering_crush_mandatory_member = 0;
```

Comment:
I don't think you want to use 0 as default/null here.  osd.0 is actually valid.  CRUSH_ITEM_NONE seems like it would be the right answer.